src/plugins/orangehrmBuzzPlugin/Dao/BuzzLikeDao.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA
+ */
+
+namespace OrangeHRM\Buzz\Dao;
+
+use OrangeHRM\Buzz\Dto\BuzzLikeSearchFilterParams;
+use OrangeHRM\Buzz\Traits\Service\BuzzServiceTrait;
+use OrangeHRM\Core\Dao\BaseDao;
+use OrangeHRM\Entity\BuzzLikeOnShare;
+use OrangeHRM\ORM\QueryBuilderWrapper;
+
+class BuzzLikeDao extends BaseDao
+{
+    use BuzzServiceTrait;
+
+    /**
+     * @param BuzzLikeOnShare $buzzLikeOnShare
+     * @return BuzzLikeOnShare
+     */
+    public function saveBuzzLikeOnShare(BuzzLikeOnShare $buzzLikeOnShare): BuzzLikeOnShare
+    {
+        $this->persist($buzzLikeOnShare);
+        return $buzzLikeOnShare;
+    }
+
+    /**
+     * @param int $shareId
+     * @param int $empNumber
+     * @return int
+     */
+    public function deleteBuzzLikeOnShare(int $shareId, int $empNumber): int
+    {
+        $qb = $this->createQueryBuilder(BuzzLikeOnShare::class, 'shareLike');
+        $qb->delete()
+            ->andWhere($qb->expr()->eq('shareLike.share', ':shareId'))
+            ->andWhere($qb->expr()->eq('shareLike.employee', ':empNumber'))
+            ->setParameter('shareId', $shareId)
+            ->setParameter('empNumber', $empNumber);
```

Comment:
We have to reduce `number_of_likes` in `ohrm_buzz_share` by 1 when deleting like no ?

Reply:
Nice catch. @devishke-orange refer 

- https://github.com/orangehrm/orangehrm/pull/1535/files#diff-059e542f32bb7cdac65aaa45dfad0a2bbed812979b77a58a333190cce849779aR170-R172
- https://github.com/orangehrm/orangehrm/pull/1535/files#diff-059e542f32bb7cdac65aaa45dfad0a2bbed812979b77a58a333190cce849779aR269-R271

Reply:
decrease likes added. please check @ManulMax @RajithaKumara https://github.com/orangehrm/orangehrm/pull/1537/commits/ddb33fcd4282ff7ce8a100571235a3876a2bf9e2#diff-7b1839d3b3930ac1cd5e7d6a8166de16dc894ab1525c11ce35b827a3cfc61f5dR178-R179